ruby/ql/lib/codeql/ruby/dataflow/internal/TaintTrackingPrivate.qll

```diff
@@ -24,12 +24,56 @@ predicate defaultTaintSanitizerGuard(DataFlow::BarrierGuard guard) { none() }
 bindingset[node]
 predicate defaultImplicitTaintRead(DataFlow::Node node, DataFlow::Content c) { none() }
 
+private CfgNodes::ExprNodes::VariableWriteAccessCfgNode variablesInPattern(
+  CfgNodes::ExprNodes::CasePatternCfgNode p
+) {
+  result = p
+  or
+  exists(CfgNodes::ExprNodes::AsPatternCfgNode ap | p = ap |
```

Comment:
I prefer `p = any(CfgNodes::ExprNodes::AsPatternCfgNode ap | ...)` (and same for the other cases).